src/main/java/org/cristalise/restapi/ScriptUtils.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * This file is part of the CRISTAL-iSE REST API.
+ * Copyright (c) 2001-2016 The CRISTAL Consortium. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation,
+ * Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ *
+ * http://www.fsf.org/licensing/licenses/lgpl.html
+ */
+package org.cristalise.restapi;
+
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
+import java.net.URLDecoder;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.Semaphore;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.cristalise.kernel.common.InvalidDataException;
+import org.cristalise.kernel.entity.proxy.ItemProxy;
+import org.cristalise.kernel.persistency.outcome.Outcome;
+import org.cristalise.kernel.persistency.outcome.Schema;
+import org.cristalise.kernel.scripting.Script;
+import org.cristalise.kernel.scripting.ScriptingEngineException;
+import org.cristalise.kernel.utils.CastorHashMap;
+import org.cristalise.kernel.utils.LocalObjectLoader;
+import org.cristalise.kernel.utils.Logger;
+import org.json.JSONObject;
+import org.json.XML;
+
+public class ScriptUtils extends ItemUtils {
```

Comment:
Why does it extend ItemUtils?

Reply:
I did that to use the produceJSON(). Should I just instantiate the ItemUtils?

Reply:
That's a static method, no need to inherit it.
If you want to make the whole code cleaner, you could move `produceJSON` and `createWebAppException` (both static) into a separate `CommonUtils` class.

Reply:
Okay. What could I do with the protected getOutcomeResponse in ItemUtils?

Reply:
Entered new issue #56 for this.

Reply:
Okay